File: 01_Language/01_Functions/java/src/main/java/com/onevgo/functions/Base64Encode.java
package com.onevgo.functions;

import cn.hutool.core.codec.Base64;

public class Base64Encode {
    public static String base64Encode(byte[] data) {
        return Base64.encode(data);
    }

    public static void main(String[] args) {
        System.out.println(base64Encode("This is an encoded string".getBytes()));
    }
}

File: 01_Language/01_Functions/python/base64_encode.py
# coding: utf-8

import sys


def base64_encode(data):
    if sys.version_info.major == 2:
        # python = 2.x:
        return data.encode('base64')
    elif sys.version_info <= (3, 1):
        # python = 3.1:
        import base64
        return base64.encodestring(data)
    else:
        # python > 3.1
        import base64
        return base64.b64encode(data.encode("utf-8")).decode("utf-8")


if __name__ == '__main__':
    print(base64_encode("This is an encoded string"))
